pwiz_tools/Skyline/TestPerf/DetectionsPlotTest.cs

```diff
@@ -0,0 +1,228 @@
+﻿using System;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using System.Windows.Forms;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using pwiz.Common.Collections;
+using pwiz.SkylineTestUtil;
+using pwiz.Skyline.Controls.Graphs;
+using pwiz.Skyline.Model;
+using pwiz.Skyline.Properties;
+using pwiz.Skyline.Util.Extensions;
+using ZedGraph;
+
+namespace TestPerf
+{
+    [TestClass]
+    public class DetectionsPlotTest : AbstractFunctionalTestEx
+    {
+        private static readonly int[][] REF_DATA =
+        {
+            new[] {132, 131, 131, 130, 130, 132},       //q = 0.003, Peptides
+            new[] { 142, 141, 141, 140, 140, 141},      //q = 0.003, Precursors
+            new[] { 129, 127, 128, 127, 127, 128 },      //q = 0.001, Peptides
+            new[] { 139, 137, 138, 137, 136, 136 },      //q = 0.001, Precursors
+            new[] { 128, 126, 127, 126, 126, 127 },      //q = 0.001, Peptides, after update
+            new[] { 138, 136, 137, 136, 135, 135 }      //q = 0.001, Precursors, after update
+        };
+
+        private static readonly string[] TIP_TEXT =
+        {
+            Resources.DetectionPlotPane_Tooltip_Replicate + TextUtil.SEPARATOR_TSV_STR + @"2_SW-B",
+            string.Format(Resources.DetectionPlotPane_Tooltip_Count, DetectionsGraphController.TargetType.PRECURSOR) + 
+            TextUtil.SEPARATOR_TSV_STR + 137.ToString( CultureInfo.CurrentCulture),
+            Resources.DetectionPlotPane_Tooltip_CumulativeCount + TextUtil.SEPARATOR_TSV_STR + 
+            143.ToString( CultureInfo.CurrentCulture),
+            Resources.DetectionPlotPane_Tooltip_AllCount + TextUtil.SEPARATOR_TSV_STR + 
+            133.ToString( CultureInfo.CurrentCulture),
+            Resources.DetectionPlotPane_Tooltip_QMedian + TextUtil.SEPARATOR_TSV_STR + 
+            6.0f.ToString(@"F1",CultureInfo.CurrentCulture)
+        };
+        private static readonly string[] HISTOGRAM_TIP_TEXT =
+        {
+            Resources.DetectionHistogramPane_Tooltip_ReplicateCount + TextUtil.SEPARATOR_TSV_STR + 
+            5.ToString( CultureInfo.CurrentCulture),
+            String.Format(Resources.DetectionHistogramPane_Tooltip_Count, DetectionsGraphController.TargetType.PRECURSOR) + 
+            TextUtil.SEPARATOR_TSV_STR + 119.ToString( CultureInfo.CurrentCulture),
+        };
+
+        [TestMethod]
+        public void TestDetectionsPlot()
+        {
+            RunPerfTests = true;

```

Comment:
Make sure you do not check this line in.

Reply:
Thanks, I commented it out.